main.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"net/url"
+	"os"
+	"os/signal"
+	"runtime"
+	"syscall"
+	"time"
+
+	"github.com/observatorium/observatorium/internal"
+	"github.com/observatorium/observatorium/internal/proxy"
+	"github.com/observatorium/observatorium/internal/server"
+
+	"github.com/go-kit/kit/log/level"
+	"github.com/oklog/run"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/version"
+	"go.uber.org/automaxprocs/maxprocs"
+)
+
+type options struct {
+	debugMutexProfileFraction int
+	debugBlockProfileRate     int
+
+	proxyBufferSizeBytes int
+	proxyBufferCount     int
+
+	listen               string
+	gracePeriod          string
+	debugName            string
+	logLevel             string
+	logFormat            string
+	metricsQueryEndpoint string
+	metricsWriteEndpoint string
+}
+
+func main() {
+	opts := options{}
+
+	flag.StringVar(&opts.listen, "listen", ":8080", "The address on which internal server runs.")
+	flag.StringVar(&opts.gracePeriod, "grace-period", "5s", "The time to wait after an OS interrupt received.")
+	flag.StringVar(&opts.debugName, "debug.name", "observatorium", "The name to add as prefix to log lines.")
+	flag.IntVar(&opts.debugMutexProfileFraction, "debug.mutex-profile-fraction", 10,
+		"The parameter which controls the fraction of mutex contention events that are reported in the mutex profile.")
+	flag.IntVar(&opts.debugBlockProfileRate, "debug.block-profile-rate", 10,
+		"The parameter controls the fraction of goroutine blocking events that are reported in the blocking profile.")
+	flag.StringVar(&opts.logLevel, "log.level", "info", "The log filtering level. Options: 'error', 'warn', 'info', 'debug'.")
+	flag.StringVar(&opts.logFormat, "log.format", internal.LogFormatLogfmt, "The log format to use. Options: 'logfmt', 'json'.")
+	flag.StringVar(&opts.metricsQueryEndpoint, "metrics-query-endpoint", "", "The endpoint which to make queries against for metrics.")
+	flag.StringVar(&opts.metricsWriteEndpoint, "metrics-write-endpoint", "", "The endpoint which to make write requests against for metrics.")
```

Comment:
Thinking we may want to namespace these flags, e.g.:

```
--metrics.query-endpoint
--metrics.write-endpoint
```
or
```
--endpoint.metrics-query
--endpoint.metrics-write
```

Reply:
I'll go with the first one, maybe we should even go with:
```
--metrics.query.endpoint
--metrics.write.endpoint
```
Just being presumptions here.

What do you think?

Reply:
that sgtm. that will allow us to eventually add additional flags, e.g. auth/tls flags, to each area separately